### PR TITLE
Fix additive attention mask handling in the native NPU attention backend.

### LINKED ISSUE #14397: Native NPU attention incorrectly converts additive attention masks to boolean masks
### Describe the bug

I found this issue while testing `dg845/LTX-2.3-Diffusers` with the [verl-omni LTX-2.3 NPU training script](https://github.com/verl-project/verl-omni/blob/main/examples/flowgrpo_trainer/ltx2/run_ltx2_3_t2av_lora_npu.sh).

This issue has only been observed and validated with LTX-2.3. During testing, the train-inference consistency error was around `1e-3`, while the usual empirical value in verl-omni is around `1e-5`. Further investigation traced the discrepancy to the handling of floating-point attention masks in `_native_npu_attention`.

Currently, every non-`None` attention mask is passed to `_maybe_modify_attn_mask_npu`. This helper assumes that the input is a boolean keep mask, converts it to `torch.bool`, and then inverts it because PyTorch SDPA and `npu_fusion_attention` use opposite boolean-mask polarities:

- PyTorch SDPA: `True` means attend.
- `npu_fusion_attention`: `True` means discard.

However, floating-point masks in PyTorch SDPA are additive biases:

- `0.0` means attend without changing the attention score.
- `-10000.0` or `-inf` means discard.

Casting an additive mask to boolean and then inverting it reverses its semantics:

```text
Additive mask:       [0.0, 0.0, -10000.0, -10000.0]
Current NPU result:  [True, True, False, False]
Expected block mask: [False, False, True, True]
```

As a result, valid positions may be discarded while masked positions may be attended, causing a silent train-inference consistency error instead of a runtime failure.

Affected code:

https://github.com/huggingface/diffusers/blob/main/src/diffusers/models/attention_dispatch.py

### Proposed fix

- Route floating-point additive masks through the existing PyTorch SDPA path.
- Keep boolean masks on the `npu_fusion_attention` path.
- Reject unsupported mask dtypes instead of silently converting them to boolean.

### Reproduction

The following isolated example demonstrates the incorrect conversion in the current helper. After the proposed routing change, floating-point masks will not be passed to this helper.

```python
import torch

from diffusers.models.attention_dispatch import _maybe_modify_attn_mask_npu


query = torch.zeros(1, 2, 1, 8)
key = torch.zeros(1, 4, 1, 8)

# PyTorch SDPA additive mask:
# 0.0 = attend, -10000.0 = discard
additive_mask = torch.tensor([[[[0.0, 0.0, -10000.0, -10000.0]]]])

actual = _maybe_modify_attn_mask_npu(query, key, additive_mask)
expected = torch.tensor(
    [[[[False, False, True, True], [False, False, True, True]]]]
)

print("Actual:")
print(actual)

print("Expected:")
print(expected)

torch.testing.assert_close(actual, expected)
```

The assertion fails because the current result is:

```text
[[[[ True, True, False, False],
   [ True, True, False, False]]]]
```

### System Info

- OS: Linux aarch64
- Hardware: Ascend NPU
- Python: 3.11
- PyTorch: 2.10.0
- torch-npu: 2.10.0
- CANN: 9.0.0
- diffusers: `main`

### Who can help?

@DN6 @yiyixuxu

### PR DESCRIPTION
**Pull request by mengchengTang**
# What does this PR do?

Fixes attention mask handling in the native NPU attention backend.

PyTorch SDPA defines two mask formats:

- Boolean masks are keep masks, where `True` means attend.
- Floating-point masks are additive biases applied directly to attention scores.

Previously, `_native_npu_attention` passed every mask to `_maybe_modify_attn_mask_npu`, which casts the mask to boolean and inverts it for `npu_fusion_attention`. This works for boolean masks but loses the values and semantics of floating-point additive masks.

This PR makes the mask handling explicit:

- Floating-point masks are routed through the existing native SDPA implementation, preserving their additive values.
- Boolean masks continue using `npu_fusion_attention`.
- Other mask dtypes are rejected with a clear error instead of being silently converted to boolean.

No new dependencies are introduced.

Fixes #14397

### Test results

| Scope | Test | Before the fix | After the fix | Result |
| --- | --- | --- | --- | --- |
| End-to-end | [`dg845/LTX-2.3-Diffusers` with verl-omni NPU training](https://github.com/verl-project/verl-omni/blob/main/examples/flowgrpo_trainer/ltx2/run_ltx2_3_t2av_lora_npu.sh) | `log_prob_diff` was around `1e-3` during the first 5 training steps | `log_prob_diff` decreased to around `1e-5` during the first 5 training steps | ✓ Passed |
| Minimal reproduction | Floating-point additive mask reproduction from this issue | The mask semantics were reversed and the assertion failed | The additive mask is handled correctly and the assertion passes | ✓ Passed |

## Before submitting
- [x] Did you use an AI agent (Claude Code, Codex, Cursor, etc.) to help with this PR? If so:
  - [x] Did you read the [Coding with AI agents](https://huggingface.co/docs/diffusers/main/en/conceptual/contribution#coding-with-ai-agents) guide?
  - [x] Did you run the [`self-review`](https://github.com/huggingface/diffusers/blob/main/.ai/skills/self-review/SKILL.md) skill on the diff?
  - [x] Did you share the final self-review notes in the PR description or a comment?
- [x] Did you read the [contributor guideline](https://huggingface.co/docs/diffusers/main/en/conceptual/contribution)?
- [x] Did you read our [philosophy doc](https://huggingface.co/docs/diffusers/main/en/conceptual/philosophy)? (important for complex PRs)
- [x] Was this discussed/approved via a GitHub issue or the [forum](https://discuss.huggingface.co/c/discussion-related-to-httpsgithubcomhuggingfacediffusers/63)? Please add a link to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/diffusers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?
- [ ] Are you the author (or part of the team) of the model/pipeline (only applicable for model/pipeline related PRs)?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by mengchengTang** [comment]
@yiyixuxu @sayakpaul @zhtmike These two Ascend NPU fixes (#14288 and #14396) are issues I ran into while adapting verl-omni — could you take a look when you get a chance? Thanks!

**Comment by zhtmike** [comment]
It is just my personal opinion, seems it is little bit strange of this fixing, it is supposed to fix mask in npu attention instead of routing to sdpa?

And kindly ping @sayakpaul, can you take a look of this PR if you have time? Seems there is some mask handling probelm in NPU attention, and this fixing is limited in NPU scope. Thanks!

File: src/diffusers/models/attention_dispatch.py [hunk 1]
```diff
@@ -3773,6 +3776,19 @@ def _native_npu_attention(
...
+        # Preserve additive mask values by letting native SDPA select the appropriate NPU implementation.
+        return _native_attention(
+            query,
+            key,
+            value,
+            attn_mask,
+            dropout_p,
+            scale=scale,
+            _parallel_config=_parallel_config,
+        )
```
Comment (sayakpaul) [inline src/diffusers/models/attention_dispatch.py:3790]: If it's NPU specific then we should add a conditional to handle it. @zhtmike could you confirm if it works as expected on CUDA?

Comment (zhtmike) [reply src/diffusers/models/attention_dispatch.py:3790]: Hi @sayakpaul, I believe the code modification is within npu scope.

the changes only takes effective only when

`transformer.set_attention_backend("_native_npu")` -> routing to `_native_npu_attention` -> fix the mask handling in NPU.

So it should be safe for CUDA.

Comment (zhtmike) [reply src/diffusers/models/attention_dispatch.py:3790]: But a kind of strange why routing to spda instead fixing the mask directly? @mengchengTang

Comment (mengchengTang) [reply src/diffusers/models/attention_dispatch.py:3790]: @zhtmike  The `npu_fusion_attention` called by `_native_npu` only supports boolean masks. Routing additive masks to SDPA prevents logic errors from forcibly converting them to boolean masks. SDPA can correctly handle additive masks.
